util/caller/resolver.go

```diff
@@ -45,8 +45,20 @@ type CallResolver struct {
 }
 
 // defaultPattern strips src/github.com/organization/project/module/submodule/file.go
-// to module/submodule/file.go.
-var defaultPattern = "src/" + strings.Repeat(strings.Join([]string{"[^", "]+", ""}, string(os.PathSeparator)), 3) + "(.*)"
+// down to module/submodule/file.go.
+var defaultPattern = func() string {
+	_, file, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("unable to look up location")
+	}
+	const sep = string(os.PathSeparator)
+	path := filepath.Dir(file)
+	// Strip to $GOPATH/src.
+	for i := 0; i < 5; i++ {
+		path = filepath.Dir(filepath.Clean(path))
+	}
```

Comment:
Panic if `path` does not end in `/src` at this point (so we can detect if this file is ever moved to a different depth in the package tree).